cmd/operator_install.go

```diff
@@ -0,0 +1,266 @@
+/*
+Copyright © 2022 NAME HERE <EMAIL ADDRESS>
+*/
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"path/filepath"
+	"sigs.k8s.io/yaml"
+	"spot-oceancd-cli/pkg/oceancd"
+	"spot-oceancd-cli/pkg/oceancd/model/operator"
+	"spot-oceancd-cli/pkg/utils"
+	"spot-oceancd-operator-commons/configs"
+	"spot-oceancd-operator-commons/handlers/cluster"
+	"spot-oceancd-operator-commons/helpers"
+	"strings"
+)
+
+var operatorManagerConfig string
+
+// operatorInstallCmd represents the operator install command
+var (
+	operatorInstallDescription      = `Installs Ocean CD operator on current cluster with dependencies based on provided config.`
+	operatorInstallShortDescription = "Installs Ocean CD operator on current cluster"
+	operatorInstallUse              = "install"
+	operatorInstallExample          = fmt.Sprintf("  # %s\n  %s %s %s %s",
+		operatorInstallShortDescription, rootCmd.Name(), operatorUse, operatorInstallUse, "--config /path/to/config")
+
+	operatorInstallCmd = &cobra.Command{
+		Use:     operatorInstallUse,
+		Short:   operatorInstallShortDescription,
+		Long:    operatorInstallDescription,
+		Example: operatorInstallExample,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			validateToken(context.Background())
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return validateOperatorInstallFlags(cmd)
+		},
+		Args: func(cmd *cobra.Command, args []string) error {
+			return cobra.NoArgs(cmd, args)
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := runOperatorInstallCmd(context.Background(), cmd); err != nil {
+				fmt.Printf("failed to install operator: %s\n", err)
+			}
+		},
+	}
+)
+
+func init() {
+	operatorCmd.AddCommand(operatorInstallCmd)
+
+	// Here you will define your flags and configuration settings.
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	// operatorCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// operatorCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	operatorInstallCmd.Flags().StringVarP(&operatorManagerConfig, "config", "c", "",
+		"The configuration applied to OceanCD resources and their dependencies.")
+
+}
+
+func runOperatorInstallCmd(ctx context.Context, cmd *cobra.Command) error {
+	pathToConfig, err := cmd.Flags().GetString("config")
+	if err != nil {
+		return fmt.Errorf("failed to parse --config flag: %w", err)
+	}
+
+	if pathToConfig != "" {
+		fmt.Printf("the %s config is being used.\n", pathToConfig)
+		configHandler, err := utils.NewConfigHandler(pathToConfig)
```

Comment:
I think ConfigHandler is not best suited here because we only have Map and not list of Maps as other commands have

I suggest either (because we want our code to be consistent and I can understand why using ConfigHandlers can help with that) change ConfigHandler to support both config path and config data this will improve the readability and simplify the code here

or you can do the deserialize to map here and then call installOperator once with either the serialized config or the default config

Reply:
as the issue is resolved by adding the config's options, let's think if we need to do this